src/workflow/history_update.rs

```diff
@@ -0,0 +1,151 @@
+use crate::{
+    protos::temporal::api::enums::v1::EventType,
+    protos::temporal::api::history::v1::{History, HistoryEvent},
+    protosext::ValidPollWFTQResponse,
+};
+use core::convert::From;
+use std::collections::VecDeque;
+
+/// A slimmed down version of a poll workflow task response which includes just the info needed
+/// by [WorkflowManager]. History events are expected to be consumed from it and applied to the
+/// state machines.
+#[derive(Debug, Clone)]
+pub struct HistoryUpdate {
+    events: VecDeque<HistoryEvent>,
+    pub previous_started_event_id: i64,
+    pub started_event_id: i64,
+}
+
+impl HistoryUpdate {
+    pub fn new(
+        history: History,
+        previous_wft_started_id: i64,
+        current_wft_started_id: i64,
+    ) -> Self {
+        Self {
+            events: history.events.into(),
+            previous_started_event_id: previous_wft_started_id,
+            started_event_id: current_wft_started_id,
+        }
+    }
+
+    #[cfg(test)]
+    pub fn new_from_events(
+        events: Vec<HistoryEvent>,
+        previous_wft_started_id: i64,
+        current_wft_started_id: i64,
+    ) -> Self {
+        Self {
+            events: events.into(),
+            previous_started_event_id: previous_wft_started_id,
+            started_event_id: current_wft_started_id,
+        }
+    }
+    /// Given a workflow task started id, return all events starting at that number (inclusive) to
+    /// the next WFT started event (inclusive). If there is no subsequent WFT started event,
+    /// remaining history is returned.
+    ///
+    /// Events are *consumed* by this process, to keep things efficient in workflow machines
+    pub fn take_next_wft_sequence(&mut self, from_wft_started_id: i64) -> Vec<HistoryEvent> {
+        let mut events_to_next_wft_started = vec![];
+        let mut saw_next_wft = false;
+
+        let mut should_pop = |e: &HistoryEvent| {
```

Comment:
nit. Could we add a comment explaining what this flag is as well as if branches below?

Reply:
Added